### PR TITLE
nav text fixup

### PR DESCRIPTION
**Pull request by mooreds**
This is a large number of small changes.

* marking whatever the 'top' file should be in a category (most of the changes) using `topOfNav`
* title renames and casing fixes
* moving some files to their directory index where it made sense (and doing the redirect)
* documenting `topOfNav`

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205962906117812

### CODE REVIEW

File: astro/src/content/docs/customize/email-and-messages/deprecated/index.mdx [hunk 1]
```diff
@@ -4,6 +4,7 @@ description: Learn how to complete Two Factor authentication for a User
 section: customize
 subcategory: email and messages
 tertcategory: deprecated
+topOfNav: true
```
Comment (mark-robustelli) [inline astro/src/content/docs/customize/email-and-messages/deprecated/index.mdx:7]: This page is labeled as an index page and should be an overview, but it is for Two Factor Auth.  I don't think the topOfNav should apply here and could be renamed.  Would need a default page then.

File: astro/src/content/docs/customize/look-and-feel/cli.mdx [hunk 2]
```diff
@@ -3,6 +3,7 @@ title: FusionAuth CLI
 description: Learn about the FusionAuth admin CLI tool
 section: customize
 subcategory: look and feel
+# TODO: move this to top level customize as it can be used for email template management too
```
Comment (mark-robustelli) [inline astro/src/content/docs/customize/look-and-feel/cli.mdx:6]: Did you mean to do this, or just that it needs to be done in the future?  I think it is the second but just want to verify.

File: astro/src/content/docs/customize/look-and-feel/index.mdx [hunk 3]
```diff
@@ -3,6 +3,7 @@ title: Themes
 description: Learn how to theme and the FusionAuth login pages (including forgot password, two-factor authentication and others)
 section: customize
 subcategory: look and feel
+topOfNav: true
```
Comment (mark-robustelli) [inline astro/src/content/docs/customize/look-and-feel/index.mdx:6]: Same comment about this not really being an overview\index page.  It is the Theme page.

File: astro/src/content/docs/customize/look-and-feel/kickstart-custom-theme.mdx [hunk 4]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Kickstart instance with custom theme
+title: Custom Themes And Kickstart
```
Comment (mark-robustelli) [inline astro/src/content/docs/customize/look-and-feel/kickstart-custom-theme.mdx:2]: Might make more sense to title it 'Theme Kickstarts' as the others.

File: astro/src/content/docs/extend/code/lambdas/index.mdx [hunk 5]
```diff
@@ -4,6 +4,7 @@ description: An overview of the lambda support of FusionAuth
 section: extend
 subcategory: code
 tertcategory: lambdas
+topOfNav: true
```
Comment (mark-robustelli) [inline astro/src/content/docs/extend/code/lambdas/index.mdx:7]: Nothing wrong with this code.  In general this page title does not follow the same format as customization.  This only has Lamdas where the Customization has 'Overview' appended.

Screenshot 2023-11-17 at 11 18 47 AM

Screenshot 2023-11-17 at 11 20 41 AM

File: astro/src/content/docs/extend/code/password-hashes/custom-password-hashing.mdx [hunk 6]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Custom Password Hashing (Password Encryptors)
+title: Custom Password Hashing
```
Comment (mark-robustelli) [inline astro/src/content/docs/extend/code/password-hashes/custom-password-hashing.mdx:2]: I'm getting an error when trying to view this page
http://localhost:3000/docs/extend/code/password-hashes/custom-password-hashing

Error: Cannot find module '@fusionauth/astro-components' imported from ....

File: astro/src/content/docs/extend/code/password-hashes/index.mdx [hunk 7]
```diff
@@ -4,6 +4,7 @@ description: Extend the functionality of FusionAuth with plugins
 section: extend
 subcategory: code
 tertcategory: password hashes
+topOfNav: true
```
Comment (mark-robustelli) [inline astro/src/content/docs/extend/code/password-hashes/index.mdx:7]: same comment about not being an actual index or overview page.

File: astro/src/content/docs/extend/examples/api-gateways/index.mdx [hunk 8]
```diff
@@ -5,6 +5,7 @@ navcategory: developer
 section: extend
 subcategory: examples
 tertcategory: api gateways
+topOfNav: true
```
Comment (mark-robustelli) [inline astro/src/content/docs/extend/examples/api-gateways/index.mdx:8]: The navigation reads only API Gateways, not API Gateways Overview

File: astro/src/content/docs/lifecycle/authenticate-users/oauth/index.mdx [hunk 9]
```diff
@@ -5,6 +5,7 @@ navcategory: login-methods
 section: lifecycle
 subcategory: authenticate users
 tertcategory: oauth
+topOfNav: true
```
Comment (mark-robustelli) [inline astro/src/content/docs/lifecycle/authenticate-users/oauth/index.mdx:8]: This page is an overview but is not marked as such in the nav.
Screenshot 2023-11-17 at 11 30 35 AM

File: astro/src/content/docs/lifecycle/authenticate-users/passwordless/index.mdx [hunk 10]
```diff
@@ -1,10 +1,11 @@
 ---
-title: Passwordless
+title: Passwordless Login
```
Comment (mark-robustelli) [inline astro/src/content/docs/lifecycle/authenticate-users/passwordless/index.mdx:2]: Not listed as Oveview in nav

File: astro/src/content/docs/lifecycle/manage-users/account-management/index.mdx [hunk 11]
```diff
@@ -5,6 +5,7 @@ navcategory: premium
 section: lifecycle
 subcategory: manage users
 tertcategory: account management
+topOfNav: true
```
Comment (mark-robustelli) [inline astro/src/content/docs/lifecycle/manage-users/account-management/index.mdx:8]: not listed as overview in nav

File: astro/src/content/docs/lifecycle/migrate-users/connectors/index.mdx [hunk 12]
```diff
@@ -1,10 +1,11 @@
 ---
-title: FusionAuth Connectors
+title: Connectors
```
Comment (mark-robustelli) [inline astro/src/content/docs/lifecycle/migrate-users/connectors/index.mdx:2]: Not listed as Overview in nav

File: astro/src/content/docs/lifecycle/migrate-users/scim/index.mdx [hunk 13]
```diff
@@ -5,6 +5,7 @@ navcategory: premium
 section: lifecycle
 subcategory: migrate users
 tertcategory: scim
+topOfNav: true
```
Comment (mark-robustelli) [inline astro/src/content/docs/lifecycle/migrate-users/scim/index.mdx:8]: not listed as overview in nav

**Review by mark-robustelli — CHANGES_REQUESTED** [review]
I don't think there was anything that had to be changed, but there are some inconsistencies in the naming and nav.  I don't think they have to be changed to push, but we should probably decide what we want and go back and make them consistent. Also made a suggestion on one of the titles.

File: astro/src/content/docs/extend/code/password-hashes/custom-password-hashing.mdx (hunk 6 above)
Comment (lyleschemmerling) [reply astro/src/content/docs/extend/code/password-hashes/custom-password-hashing.mdx:2]: You need to run `npm install` to pick up the library

Comment (mark-robustelli) [reply astro/src/content/docs/extend/code/password-hashes/custom-password-hashing.mdx:2]: Makes sense.  Thx.

File: astro/src/content/docs/customize/look-and-feel/cli.mdx (hunk 2 above)
Comment (mooreds) [reply astro/src/content/docs/customize/look-and-feel/cli.mdx:6]: Need to be done in the future. Didn't want too many moving pieces in this PR.

File: astro/src/content/docs/customize/look-and-feel/kickstart-custom-theme.mdx (hunk 4 above)
Comment (mooreds) [reply astro/src/content/docs/customize/look-and-feel/kickstart-custom-theme.mdx:2]: What does 'Theme Kickstarts' mean? 

The point of this article is that you can use kickstart to set up themes. How about `Setting Up A Theme with Kickstart`?

File: astro/src/content/docs/extend/code/lambdas/index.mdx (hunk 5 above)
Comment (mooreds) [reply astro/src/content/docs/extend/code/lambdas/index.mdx:7]: Yes, the titles are a hot mess and not consistent. I fixed a few where they sprawled across the page, but we should definitely make them consistent w/r/t title casing and whether to put 'Overview' on it or not.

Not sure that should go into this PR.

File: astro/src/content/docs/extend/examples/api-gateways/index.mdx (hunk 8 above)
Comment (mooreds) [reply astro/src/content/docs/extend/examples/api-gateways/index.mdx:8]: I'm not sure I understand.

**Review by mark-robustelli — APPROVED** [review]
LGTM

**Pull request merged**